addons/product_configurator/static/src/js/product_configurator_field.js

```diff
@@ -0,0 +1,152 @@
+                }
+            }
+        } else {
+            if (!result.mode || result.mode === 'configurator') {
+                this._openProductConfigurator();
+            } else {
+                // only triggered when sale_product_matrix is installed.
+                this._openGridConfigurator();
+            }
+        }
+    },
+
+    async _openProductConfigurator(edit=false) {
+        /**
+         *  `product_custom_attribute_value_ids` records are not loaded in the view bc sub templates
+         *  are not loaded in list views. Therefore, we fetch them from the server if the record is
+         *  saved. Else we use the value stored on the line.
+         */
+        const params = await this.configuratorParams(edit)
+        this.dialog.add(ProductConfiguratorDialog, params);
+    },
+
+    async configuratorParams(edit){
+        const customAttributeValues =
+            this.props.record.data.product_custom_attribute_value_ids.records[0]?.isNew ?
+                this.props.record.data.product_custom_attribute_value_ids.records.map(
+                    record => record.data
+                ) :
+                await this.orm.read(
+                    'product.attribute.custom.value',
+                    this.props.record.data.product_custom_attribute_value_ids.currentIds,
+                    ["custom_product_template_attribute_value_id", "custom_value"]
+                );
+        return {
+            productTemplateId: this.props.record.data.product_tmpl_id[0],
+            ptavIds: this.props.record.data.product_template_attribute_value_ids.records.map(
+                record => record.resId
+            ).concat(this.props.record.data.product_no_variant_attribute_value_ids.records.map(
+                record => record.resId
+            )),
+            customAttributeValues: customAttributeValues.map(
+                data => {
+                    return {
+                        ptavId: data.custom_product_template_attribute_value_id[0],
+                        value: data.custom_value,
+                    }
+                }
+            ),
+            edit: edit,
+            options: this.props.options,
+            save: async (mainProduct, optionalProducts) => {
+                await this.applyProduct(this.props.record, mainProduct);
+            },
+            discard: () => {},
+            ...this.additionalParams(),
+        };
+    },
+
+    additionalParams() {
+        if(this.props.options?.hideQty){
+            return {companyId: this.props.record.model.root.data.company_id[0]};
+        }
+        const product_uom_id = this.props.options?.productUomIdFieldName ? this.props.options.productUomIdFieldName : 'product_uom';
+        return {
+            quantity: this.props.record.data.product_uom_qty,
+            productUOMId: this.props.record.data[product_uom_id][0],
+            companyId: this.props.record.model.root.data.company_id[0]
+        };
+    },
```

Comment:
This won't fit. Each module needs to be able to define at least a `save` method. And currently, SaleProductField doesn't even use that so why bother dividing it into three methods?